### PR TITLE
+ update profile

### PR DESCRIPTION
**Pull request by giapnguyenitf**
https://dev.framgia.com/redmine/issues/157464
https://dev.framgia.com/redmine/issues/157465

### CODE REVIEW

File: resources/views/home.blade.php [hunk 1]
```diff
@@ -1,3 +1,101 @@
...
+                    <h4 class="caption subtitle">Dual SIM</h4>
+                    <a class="caption button-radius" href="#">
+                        <span class="icon"></span>Shop now</a>
+                </div>
+            </li>
+            <li>
+                <img src="{{ asset('images/h4-slide2.png') }}" alt="Slide">
+                <div class="caption-group">
+                    <h2 class="caption title">
+                        by one, get one
```
Comment (trangkarol) [inline resources/views/home.blade.php]: I18n

File: resources/views/home.blade.php [hunk 2]
```diff
@@ -1,3 +1,101 @@
...
+    <div class="block-slider block-slider4">
+        <ul class="" id="bxslider-home4">
+            <li>
+                <img src="{{ asset('images/h4-slide.png') }}" alt="Slide">
+                <div class="caption-group">
+                    <h2 class="caption title">
+                        iPhone
+                        <span class="primary">6
+                            <strong>Plus</strong>
+                        </span>
```
Comment (trangkarol) [inline resources/views/home.blade.php]: hard code hả  e

**Comment by trangkarol** [comment]
commented

**Comment by giapnguyenitf** [comment]
ready

File: app/Http/Controllers/User/UserController.php [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+     * Update the specified resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function update(UpdateProfileRequest $request, $id)
+    {   try
+        {
+            $gender = $request->gender == 'male' ? 1 : 0;
```
Comment (trangkarol) [inline app/Http/Controllers/User/UserController.php]: `1, 0` vào file config setting e ơi

File: resources/views/home.blade.php [hunk 4]
```diff
@@ -1,3 +1,60 @@
 @extends('layouts.master')
 @section('content')
+<div class="slider-area">
+    <!-- Slider -->
+    <div class="block-slider block-slider4">
+        <ul class="" id="bxslider-home4">
+            @foreach ($hotProducts as $hotProduct)
+                <li>
+                    <img src="{{ asset('images/h4-slide.png') }}" alt="Slide">
```
Comment (trangkarol) [inline resources/views/home.blade.php]: `alt="Slide"` nếu dùng cái này cũng phải I18n e nhé

File: resources/views/layouts/notifications.blade.php [hunk 5]
```diff
@@ -0,0 +1,22 @@
+<!-- update profile -->
+@if(Session::has('update_profile_success'))
```
Comment (trangkarol) [inline resources/views/layouts/notifications.blade.php]: Convention: `@if (Session::has('update_profile_success'))`

File: resources/views/layouts/notifications.blade.php [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    <div class="alert alert-success">
+        {{ Session::get('update_profile_success') }}
+    </div>
+@endif
+@if(Session::has('update_profile_fail'))
```
Comment (trangkarol) [inline resources/views/layouts/notifications.blade.php]: Convention: `@if (Session::has('update_profile_fail'))`

File: resources/views/layouts/notifications.blade.php [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+    <div class="alert alert-danger">
+        {{ Session::get('update_profile_fail') }}
+    </div>
+@endif
+<!-- update password -->
+@if(Session::has('update_password_success'))
```
Comment (trangkarol) [inline resources/views/layouts/notifications.blade.php]: Convention: `@if (Session::has('update_password_success'))`

File: resources/views/layouts/notifications.blade.php [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+    <div class="alert alert-success">
+        {{ Session::get('update_password_success') }}
+    </div>
+@endif
+@if(Session::has('update_password_fail'))
```
Comment (trangkarol) [inline resources/views/layouts/notifications.blade.php]: Convention: `@if (Session::has('update_password_fail'))`

File: resources/views/user/profile.blade.php [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+                {{ Form::close() }}
+            </div>
+            <hr>
+            <div class="form-update-password">
+                {{ Form::open(['route' => ['password.update', 'id' => Auth::user()->id], 'method' => 'PUT', ]) }}
+                    <div class=" row form-group{{ $errors->has('password') ? 'has-error' : '' }}">
+                        {{ Form::label('password', trans('label.password'), ['class' => 'col-md-2']) }}
+                        <div class="col-md-6">
+                            {{ Form::password('password', ['class' => 'form-control']) }}
+                            @if($errors->has('password'))
```
Comment (trangkarol) [inline resources/views/user/profile.blade.php]: Convention: `@if ($errors->has('password'))`

File: resources/views/user/profile.blade.php [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+                                    <strong>{{ $errors->first('password') }}</strong>
+                                </span>
+                            @endif
+                        </div>
+                    </div>
+                    <div class="row form-group{{ $errors->has('password_confirmation') ? 'has-error' : '' }}">
+                        {{ Form::label('password_confirmation', trans('label.password_confirmation'), ['class' => 'col-md-2']) }}
+                        <div class="col-md-6">
+                            {{ Form::password('password_confirmation', ['class' => 'form-control']) }}
+                            @if($errors->has('password_confirmation'))
```
Comment (trangkarol) [inline resources/views/user/profile.blade.php]: Convention: `@if ($errors->has('password_confirmation'))`

File: routes/web.php [hunk 11]
```diff
@@ -21,3 +21,15 @@
 
 Route::get('auth/{social}', 'Auth\SocialAuthController@redirectToProvider')->name('socialAuth');
 Route::get('auth/{social}/callback', 'Auth\SocialAuthController@handleProviderCallback');
+
+Route::prefix('user')->group(function () {
+    Route::resource('profile', 'User\UserController',  ['only' => [
```
Comment (trangkarol) [inline routes/web.php]: Covnetion: Sau dấu `,` là 1 space e ơi

**Comment by trangkarol** [comment]
commented

**Comment by giapnguyenitf** [comment]
ready

File: app/Http/Controllers/HomeController.php [hunk 12]
```diff
@@ -23,6 +26,9 @@ public function __construct()
      */
     public function index()
     {
-        return view('home');
+        $products = $this->productRepository->all(['*']);
```
Comment (trangkarol) [inline app/Http/Controllers/HomeController.php]: ở base repo e nếu có 
> function all($column = ['*']) luôn đi e 

thì e chỉ cần  `$this->productRepository->all();`

**Comment by trangkarol** [comment]
commented

**Comment by giapnguyenitf** [comment]
ready

**Pull request merged**